common/util.go

```diff
@@ -87,12 +108,41 @@ func HandleHTTPRequest(message string, headers http.Header, data ConnectorMetada
 		}
 	}
 
+	errResp := ErrorResponse{
```

Comment:
Please see if we can improve the readability of the overall code just by methods.

func NewErrorResponse() ErrorResponse 

func (err *ErrorReponse) UpdateResponseDetails()

Need better structure for code 125-145